### PR TITLE
[MiMoV2Flash] [feat]: support two batch overlap

### PR DESCRIPTION
**Pull request by TZHelloWorld**
## Motivation
support mimo_v2_flash two batch overlap:
p:
```bash
python3 -m sglang.launch_server \
    --model-path /mnt/mify-gw-model-alicn3/models/global_step_84-FP8-Block \
    --pp-size 1 --dp-size 2 --tp-size 8 \
    --enable-dp-attention \
    --moe-a2a-backend deepep \
    --deepep-mode normal \
    --disaggregation-mode prefill \
    --page-size 1 \
    --host 0.0.0.0 \
    --port 30010 \
    --trust-remote-code \
    --moe-dense-tp-size 1 \
    --enable-dp-lm-head \
    --mem-fraction-static 0.7 \
    --max-running-requests 32 \
    --reasoning-parser qwen3 \
    --tool-call-parser mimo \
    --context-length 262144 \
    --model-loader-extra-config '{"enable_multithread_load": "true","num_threads": 64}' \
    --attention-backend fa3 \
    --allow-auto-truncate \
    --chunked-prefill-size 16384 \
    --enable-two-batch-overlap
```
d:
```bash
SGLANG_DEEPEP_NUM_MAX_DISPATCH_TOKENS_PER_RANK=1024 \
python3 -m sglang.launch_server \
    --model-path /mnt/mify-gw-model-alicn3/models/global_step_84-FP8-Block \
    --pp-size 1 --dp-size 2 --tp-size 8 \
    --enable-dp-attention \
    --moe-a2a-backend deepep \
    --deepep-mode low_latency \
    --decode-log-interval 1 \
    --page-size 1 \
    --host 0.0.0.0 --port 30020 \
    --trust-remote-code \
    --watchdog-timeout 1000000 \
    --mem-fraction-static 0.7 \
    --max-running-requests 32 \
    --reasoning-parser qwen3 \
    --tool-call-parser mimo \
    --context-length 262144 \
    --model-loader-extra-config '{"enable_multithread_load": "true","num_threads": 64}' \
    --attention-backend fa3 \
    --disaggregation-mode decode \
    --moe-dense-tp-size 1 \
    --enable-dp-lm-head \
    --enable-two-batch-overlap
```
lb:
```bash
python -m sglang_router.launch_router \
        --pd-disaggregation \
        --prefill http://127.x.x.1:30010 \
        --decode http://127.x.x.1:30020 \               
        --host 0.0.0.0 \               
        --port 30000 \
        --mini-lb 
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/batch_overlap/operations_strategy.py [hunk 1]
```diff
@@ -209,3 +218,78 @@ def _compute_moe_qwen3_decode(layer):
...
+            layer.mlp.op_dispatch_a,
+            operations.YieldOperation(),  # todo, adjust
+            layer.mlp.op_dispatch_b,
+            layer.mlp.op_experts,
+            layer.mlp.op_combine_a,
+            operations.YieldOperation(),  # todo, adjust
+            layer.mlp.op_combine_b,
+            layer.mlp.op_output,
+            layer.op_comm_postprocess_layer,
+            operations.YieldOperation(),  # todo, adjust
```
Comment (acelyc111) [inline python/sglang/srt/batch_overlap/operations_strategy.py]: Leave one TODO to describe what you want to do is enough?

File: python/sglang/srt/models/mimo_v2_flash.py [hunk 2]
```diff
@@ -694,19 +866,39 @@ def forward(
...
-                hidden_states,
-                forward_batch,
-                residual,
-                captured_last_layer_outputs=(
-                    aux_hidden_states
-                    if getattr(layer, "_is_layer_to_capture", False)
-                    else None
-                ),
+        if forward_batch.can_run_tbo:
+            normal_end_layer = self.start_layer
```
Comment (acelyc111) [inline python/sglang/srt/models/mimo_v2_flash.py]: rename to tbo_start_layer?

File: python/sglang/srt/batch_overlap/operations_strategy.py [hunk 3]
```diff
@@ -209,3 +218,78 @@ def _compute_moe_qwen3_decode(layer):
...
+# -------------------------------- Strategy for MiMoV2DecoderLayer ---------------------------------------
+
+
+# TODO: unstable, current strategy is almost the same as DeepSeek, keep redundant code here for
+# convenience to adjust strategy
+def _compute_moe_mimov2_layer_operations_strategy_tbo(
+    layer: torch.nn.Module,
+    forward_mode: ForwardMode,
+) -> OperationsStrategy:
+    assert layer.is_layer_sparse, "MiMoV2DecoderLayer moe only support sparse layers"
```
[the remark this answers is not in the document]
Comment (acelyc111) [reply python/sglang/srt/batch_overlap/operations_strategy.py:232]: MiMoV2 has only 1 dense layer at layer-0.

**Review by acelyc111 — APPROVED** [review]

**Comment by Kangyan-Zhou** [comment]
@TZHelloWorld could you please resolve conflicts?
